packages/vuetify/src/components/VAppBar/VAppBar.tsx

```diff
@@ -0,0 +1,113 @@
+// Styles
+import './VAppBar.sass'
+
+// Composables
+import { makeBorderProps, useBorder } from '@/composables/border'
+import { makeDensityProps, useDensity } from '@/composables/density'
+import { makeElevationProps, useElevation } from '@/composables/elevation'
+import { makeLayoutItemProps, useLayoutItem } from '@/composables/layout'
+import { makePositionProps, usePosition } from '@/composables/position'
+import { makeRoundedProps, useRounded } from '@/composables/rounded'
+import { makeTagProps } from '@/composables/tag'
+// import { useDisplay } from '@/composables/display'
+// import { useProxiedModel } from '@/composables/proxiedModel'
+import { useTheme } from '@/composables/theme'
+import { useBackgroundColor } from '@/composables/color'
+
+// Utilities
+import { makeProps } from '@/util/makeProps'
+import { computed, defineComponent, toRef } from 'vue'
+
+export default defineComponent({
+  name: 'VAppBar',
+
+  props: makeProps({
+    color: String,
+    flat: Boolean,
+    height: [Number, String],
+    modelValue: Boolean,
+    prominent: Boolean,
+    src: String,
+    temporary: Boolean,
+    ...makeBorderProps(),
+    ...makeDensityProps(),
+    ...makeElevationProps(),
+    ...makeLayoutItemProps(),
+    ...makePositionProps(),
+    ...makeRoundedProps(),
+    ...makeLayoutItemProps({ name: 'app-bar' }),
+    ...makeTagProps({ tag: 'header' }),
+  }),
+
+  setup (props, { slots }) {
+    const { themeClasses } = useTheme()
+    const { borderClasses } = useBorder(props, 'v-app-bar')
+    const { densityClasses } = useDensity(props, 'v-app-bar')
+    const { elevationClasses } = useElevation(props)
+    // const { mobile } = useDisplay()
+    const { positionClasses, positionStyles } = usePosition(props, 'v-app-bar')
+    const { roundedClasses } = useRounded(props, 'v-app-bar')
+    const { backgroundColorClasses, backgroundColorStyles } = useBackgroundColor(toRef(props, 'color'))
+    // const isActive = useProxiedModel(props, 'modelValue')
+    const layoutStyles = useLayoutItem(
+      props.name,
+      toRef(props, 'priority'),
+      computed(() => 'top'),
```

Comment:
could be `ref('top')`

Reply:
This resulted in multiple TS errors to do with incompatible types. opting to pass for now since it's only being used in 2 components at this time.